src/pages/settings/user/components/Details.tsx

```diff
@@ -8,23 +8,33 @@
  * @license https://www.elastic.co/licensing/elastic-license
  */
 
+import { useCurrentCompany } from 'common/hooks/useCurrentCompany';
 import { useCurrentUser } from 'common/hooks/useCurrentUser';
 import { updateChanges } from 'common/stores/slices/user';
 import { RootState } from 'common/stores/store';
-import { ChangeEvent } from 'react';
+import { CustomField } from 'components/CustomField';
 import { useTranslation } from 'react-i18next';
 import { useDispatch, useSelector } from 'react-redux';
 import { Card, Element } from '../../../../components/cards';
 import { InputField } from '../../../../components/forms';
 
 export function Details() {
   const [t] = useTranslation();
+
+  const user = useCurrentUser();
+
   const dispatch = useDispatch();
+
+  const company = useCurrentCompany();
+
   const userChanges = useSelector((state: RootState) => state.user.changes);
-  const user = useCurrentUser();
-  const handleChange = (event: ChangeEvent<HTMLInputElement>) =>
+
+  const handleChange = (property: string, value: string) =>
```

Comment:
In order to prevent future type assertions, change signature to following:

```ts
const handleChange = (property: string, value: string | number | boolean)
```